CRM/Financial/BAO/FinancialType.php

```diff
@@ -345,6 +345,42 @@ public static function getAvailableMembershipTypes(&$membershipTypes = NULL, $ac
     return $membershipTypes;
   }
 
+  /**
+   * This function adds the Financial ACL clauses to the where clause.
+   *
+   * This is currently somewhat mocking the native hook implementation
+   * combined with applying the acls in core. This is seen as a transitional step
+   * as it would not need to call the hook if the financialreportsacl
+   * extension is installed. And we probably need to check if it IS
+   * installed before adding ACLs as part of the transition.
+   *
+   * @param array $whereClauses
+   */
+  public static function addACLClausesToWhereClauses(&$whereClauses) {
+    $originalWhereClauses = $whereClauses;
+    CRM_Utils_Hook::selectWhereClause('Contribution', $whereClauses);
+    if ($whereClauses !== $originalWhereClauses) {
+      // In this case permisssions have been applied & we assume the
+      // financialaclreport is applying these
+      // https://github.com/JMAConsulting/biz.jmaconsulting.financialaclreport/blob/master/financialaclreport.php#L107
+      // Long term we want the financial type status permissioning entirely removed from core.
+      return;
+    }
+
+    if (!self::isACLFinancialTypeStatus()) {
+      return;
+    }
+    $types = self::getAllEnabledAvailableFinancialTypes();
+    if (empty($types)) {
+      $whereClauses['financial_type_id'] = 'IN (0)';
+    }
+    else {
+      $whereClauses['financial_type_id'] = [
+        'IN (' . implode(',', array_keys($types)) . ')'
+      ];
+    }
```

Comment:
how about 
```php
$types = (array) self::getAllEnabledAvailableFinancialTypes();
$whereClauses['financial_type_id'] = [
        'IN (' . implode(',', array_keys($types)) . ')'
      ];
```

Reply:
My bad, silly thought. I thought array casting would have provided array_keys($types) as [0]. please ignore

Reply:
@eileenmcnaughton just for reusability, can we call here:
```diff
- $types = self::getAllEnabledAvailableFinancialTypes();
-   if (empty($types)) {
-    $whereClauses['financial_type_id'] = 'IN (0)';
-    }
-   else {
-    $whereClauses['financial_type_id'] = [
-       'IN (' . implode(',', array_keys($types)) . ')'
-    ];
-  }
+ CRM_Financial_BAO_FinancialType::buildPermissionedClause($whereClauses);
```

Reply:
@monishdeb the syntax is - buildPermissionedWhereClause doesn't follow the hook format where the field name is in the key rather than the string - so the goal is to migrate away from that to this & to move the hook compatible code eventually to being on the BAO in a way that it is generically called (e.g. by the api) as works with activities for example

Reply:
I see